tritium-lib/src/main/java/com/palantir/tritium/proxy/Instrumentation.java

```diff
@@ -143,18 +144,18 @@ private Builder(Class<T> interfaceClass, U delegate) {
          * metric names are chosen based off of the interface name and invoked method.
          *
          * @param metricRegistry - TaggedMetricsRegistry used for this application.
-         * @param globalPrefix - Metrics name prefix to be used
+         * @param prefix - Metrics name prefix to be used
          * @return - InstrumentationBuilder
          */
-        public Builder<T, U> withTaggedMetrics(TaggedMetricRegistry metricRegistry, String globalPrefix) {
+        public Builder<T, U> withTaggedMetrics(TaggedMetricRegistry metricRegistry, String prefix) {
             checkNotNull(metricRegistry, "metricRegistry");
-            this.handlers.add(new TaggedMetricsServiceInvocationEventHandler(
-                    metricRegistry, globalPrefix));
+            String serviceName = (Strings.isNullOrEmpty(prefix)) ? interfaceClass.getName() : prefix;
+            this.handlers.add(new TaggedMetricsServiceInvocationEventHandler(metricRegistry, serviceName));
             return this;
         }
 
         public Builder<T, U> withTaggedMetrics(TaggedMetricRegistry metricRegistry) {
-            return withTaggedMetrics(metricRegistry, null);
+            return withTaggedMetrics(metricRegistry, "");
```

Comment:
Oops.

Reply:
This was just a nit fix as blank would previously also have thrown as well in https://github.com/palantir/tritium/blob/develop/tritium-core/src/main/java/com/palantir/tritium/event/InstrumentationProperties.java#L42 as you need a non-null && non-blank service name.